homeassistant/components/alarm_control_panel/device_condition.py

```diff
@@ -0,0 +1,137 @@
+from homeassistant.helpers.typing import ConfigType, TemplateVarsType
+from homeassistant.helpers.config_validation import DEVICE_CONDITION_BASE_SCHEMA
+from . import DOMAIN
+
+CONDITION_TYPES = {
+    "is_triggered",
+    "is_disarmed",
+    "is_armed_home",
+    "is_armed_away",
+    "is_armed_night",
+}
+
+CONDITION_SCHEMA = DEVICE_CONDITION_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(CONDITION_TYPES),
+    }
+)
+
+
+async def async_get_conditions(
+    hass: HomeAssistant, device_id: str
+) -> List[Dict[str, str]]:
+    """List device conditions for Alarm control panel devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    conditions = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        state = hass.states.get(entry.entity_id)
+
+        # We need a state or else we can't populate the different armed conditions
+        if state is None:
+            continue
+
+        supported_features = state.attributes["supported_features"]
+
+        # Add conditions for each entity that belongs to this integration
+        conditions += [
+            {
+                CONF_CONDITION: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_ENTITY_ID: entry.entity_id,
+                CONF_TYPE: "is_disarmed",
+            },
+            {
+                CONF_CONDITION: "device",
+                CONF_DEVICE_ID: device_id,
+                CONF_DOMAIN: DOMAIN,
+                CONF_ENTITY_ID: entry.entity_id,
+                CONF_TYPE: "is_triggered",
+            },
+        ]
+        if supported_features & SUPPORT_ALARM_ARM_HOME:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "is_armed_home",
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_AWAY:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "is_armed_away",
+                }
+            )
+        if supported_features & SUPPORT_ALARM_ARM_NIGHT:
+            conditions.append(
+                {
+                    CONF_CONDITION: "device",
+                    CONF_DEVICE_ID: device_id,
+                    CONF_DOMAIN: DOMAIN,
+                    CONF_ENTITY_ID: entry.entity_id,
+                    CONF_TYPE: "is_armed_night",
+                }
```

Comment:
It looks like `SUPPORT_ALARM_ARM_CUSTOM_BYPASS` isn't accounted for.  It doesn't appear to be used in anything but 
`homeassistant/components/manual/alarm_control_panel.py:    SUPPORT_ALARM_ARM_CUSTOM_BYPASS`